tests/test_document_reference.py

```diff
@@ -31,12 +35,148 @@ def test_get_document_reference(self, get_token_client_credentials):
             headers={
                 "Accept": "application/fhir+json;version=1",
                 "Authorization": f"Bearer {token}",
-                "NHSD-End-User-Organisation-ODS": end_user_ods,
+                "NHSD-Target-Identifier": self.target_identifier_encoded,
+                "NHSD-End-User-Organisation-ODS": self.end_user_ods,
+                "X-Request-Id": "60E0B220-8136-4CA5-AE46-1D97EF59D068",
+                "X-Correlation-Id": "11C46F5F-CDEF-4865-94B2-0EE0EDCC26DA",
+            },
+        )
+        # Then
+        assert_that(expected_status_code).is_equal_to(response.status_code)
+
+    @pytest.mark.appointment
+    @pytest.mark.integration
+    @pytest.mark.sandbox
+    @pytest.mark.debug
+    def test_get_document_reference_by_id(self, get_token_client_credentials):
+        """
+           Test for the GET /DocumentReference/{id} endpoint. This operation will call the Producer NRL API.
+        """
+        document_id = "V4T0L-"
+        timestamp = int(time.time())
+        random.seed(timestamp)
+        random_integer = random.randint(0, 1000)
+        document_id += str(random_integer)
+        # Given
+        token = get_token_client_credentials["access_token"]
+        expected_status_code = 404
+        # When
+        response = requests.get(
+            url=f"{config.BASE_URL}/{config.BASE_PATH}/DocumentReference/{document_id}",
+            headers={
+                "Accept": "application/fhir+json;version=1",
+                "Authorization": f"Bearer {token}",
+                "NHSD-Target-Identifier": self.target_identifier_encoded,
+                "NHSD-End-User-Organisation-ODS": self.end_user_ods,
                 "X-Request-Id": "60E0B220-8136-4CA5-AE46-1D97EF59D068",
                 "X-Correlation-Id": "11C46F5F-CDEF-4865-94B2-0EE0EDCC26DA",
             },
         )
 
         # Then
         assert_that(expected_status_code).is_equal_to(response.status_code)
-        assert_that(expected_body).is_equal_to(response.json())
+
+    @pytest.mark.appointment
+    @pytest.mark.integration
+    @pytest.mark.sandbox
+    @pytest.mark.debug
+    def test_post_delete_put_document_reference(self, get_token_client_credentials):
+        """
+           Test for the POST, Delete and PUT /DocumentReference endpoint. This operation will call the Producer NRL API.
+        """
+        document_id = "V4T0L-"
+        timestamp = int(time.time())
+        random.seed(timestamp)
+        random_integer = random.randint(0, 1000)
+        document_id += str(random_integer)
+        # Given
+        token = get_token_client_credentials["access_token"]
+        expected_post_status_code = 201
+        expected_delete_status_code = 200
+        expected_put_status_code = 404
+        request_payload = {
```

Comment:
Just to confirm. Do we have an example of a request body in the spec? if yes then we should treat request bodies like responses i.e. loading them from a file. In this case we should load, then parse and then change the values before sending it.
If we don't have examples in the spec, please leave it for now. I think probably this is something that we need to get back to once they added more examples to the spec

Reply:
No as of now we dont have example